commit 33f79d7df61841b8987ac4a021deb7d26fc3bf61 (refs/changes/39/1239/1)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2017-01-11T18:17:50-08:00 (2 years, 9 months ago)
    
    MANTA-3084 muskie fails to read from HTTP PUT response stream: sadness ensues

diff --git a/lib/obj.js b/lib/obj.js
index 31f38cb..6686ece 100644
--- a/lib/obj.js
+++ b/lib/obj.js
@@ -520,6 +520,17 @@ function sharkStreams(req, res, next) {
                 sharkInfo.result = 'ok';
                 barrier.done(s._shark.manta_storage_id);
             }
+            /*
+             * Even though PUT requests that are successful normally result
+             * in an empty resonse body from nginx, we still need to make sure
+             * we let the response stream emit 'end'. Otherwise this will jam
+             * up keep-alive agent connections (the node http.js needs that
+             * 'end' even to happen before relinquishing the socket).
+             *
+             * Easiest thing to do is just call resume() which should make the
+             * stream run out and emit 'end'.
+             */
+            sres.resume();
         });
     });
 
